### PR TITLE
Upstream a variety of XHR tests from jsdom

### PR DESCRIPTION
**Pull request by domenic**
These have been reviewed upstream. formdata-constructor.html passes in Chrome but not Firefox or Safari, I believe because of https://github.com/whatwg/html/issues/6247 (/cc @andreubotella). Others pass in both.

### CODE REVIEW

**Comment by andreubotella** [comment]
> formdata-constructor.html passes in Chrome but not Firefox, I believe because of [whatwg/html#6247](https://github.com/whatwg/html/issues/6247) (/cc @andreubotella).

Not quite: it's because Firefox mistakenly defers the newline normalization (that in the spec happens in the "construct the entry list" algorithm) until the form payloads are constructed (https://bugzilla.mozilla.org/show_bug.cgi?id=1651887 and https://bugzilla.mozilla.org/show_bug.cgi?id=1657844). whatwg/html#6247 adds another layer of normalization at the point that form payloads are built to cover the remaining cases, including filenames in urlencoded and text/plain (since filenames don't get normalized in `multipart/form-data`).

**Comment by andreubotella** [comment]
See whatwg/html#6469 though.

File: xhr/formdata-constructor.html [hunk 1]
```diff
@@ -0,0 +1,150 @@
+<!DOCTYPE html>
+<title>Errors thrown in XMLHttpRequest events get to window.onerror</title>
```
Comment (annevk) [inline xhr/formdata-constructor.html:2]: This title doesn't match the tests.

**Review by annevk — APPROVED** [review]
Looks good, found one nit.

I guess if we make the changes around newline normalization we'll have to remember to adjust these.

**Pull request merged**